app/models/duration_rate.rb

```diff
@@ -2,10 +2,26 @@
 
 class DurationRate < ApplicationRecord
 
-  belongs_to :instrument, foreign_key: :product_id
+  belongs_to :price_group
+  belongs_to :rate_start, required: true
 
-  validates :min_duration, presence: true
-  validates :min_duration, numericality: { greater_than_or_equal_to: 0, allow_blank: true }
-  validates :rate, presence: true
+  validate :rate_or_subsidy
   validates :rate, numericality: { greater_than_or_equal_to: 0, allow_blank: true }
+  validates :subsidy, numericality: { greater_than: 0, allow_blank: true }
+
+  # TODO: Validate:
+  # - Rate is <= base rate (if Price group is the “base” group or external).
+  # - Subsidy is <= base rate (if Price group is internal but not the “base”).
+  # For this it looks like we'll need to link Rate Start to Price Policy, instead of Instrument.
```

Comment:
Totally in support of implementing this later, but wanted to share my thoughts.  Each product should only have one _current_ price policy per price group - and in the context of the controller I think we should have all the relevant data by way of the price policy object. The price policy knows its price group and the price groups knows the duration rate.  I don't think we need rate start here, do we?

Reply:
I understand what you mean.

I tried to rise 2 different questions here:
1. Business: does it make sense to have only one set of duration rates per instrument? Should I stick to a set of intervals for all my price policies? I definitely lack knowledge here, but I'm wondering whether it makes sense to say that for one price policy my intervals were 1-3, 3-5, 5+ hours, and then for my new Price policy I want them to be 2-4, 4-5, 5+
2. How to get the base rate. Now I understand I could just look for the current Price policy for the instrument and that's it 🙂